### PR TITLE
master Lps 159750 | Add tests on headless delivery discovery

### PR DESCRIPTION
**Pull request by yanan**
**Background**
- Add test to assert headless delivery api method response code in swagger
- Add test to consume graphQL in headless delivery api
- Add test to assert header works in positive case in headless delivery api
- Add test to assert header works in negative case in headless delivery api

**Test Plan**
- LPS-159748
Given Navigate to [headless-delivery](http://localhost:8080/o/api?endpoint=http://localhost:8080/o/headless-delivery/v1.0/openapi.json) endpoint
When Executing getSiteBlogPosingPage with siteID as the parameter*Then* the shown response code is 200
- LPS-159750
Given Navigate to [headless-delivery](http://localhost:8080/o/api?endpoint=http://localhost:8080/o/headless-delivery/v1.0/openapi.json) endpoint
And Given Switch to GraphQL
And Given Create a blog entry using specific title
When Executing query blog headline title*Then* Assert blog entry title equals to the one we created
- LPS-159848
Given an StructuredContent with en-US and es-ES languages included
And Given Navigate to [headless-delivery](http://localhost:8080/o/api?endpoint=http://localhost:8080/o/headless-delivery/v1.0/openapi.json) endpoint
And Given Added Accept-Language header and set value to es-ES
When Executing getSiteStructuredContentsPage with siteID as the parameter
Then Assert Accept-Language header is included in request under Curl section,
         Assert the structured content with the es-ES content is included in response
- LPS-159849
Given Navigate to [headless-delivery](http://localhost:8080/o/api?endpoint=http://localhost:8080/o/headless-delivery/v1.0/openapi.json) endpoint
And Given a site with english as default language
And Given Added Accept-Language header and set value to es-ES
When Executing postSiteStructuredContents with siteID as the parameter
Then Assert Accept-Language header is included in request under Curl section,
         BAD request is included in response section,
         Assert details message in console

**JIRA Ticket**
- https://issues.liferay.com/browse/LPS-159748
- https://issues.liferay.com/browse/LPS-159750
- https://issues.liferay.com/browse/LPS-159848
- https://issues.liferay.com/browse/LPS-159849

### CODE REVIEW

**Comment by yanan** [comment]
ci:test:headless-frontend-functional:nocompile

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Button.path [hunk 1]
```diff
@@ -990,6 +990,26 @@
 	<td>//li[contains(@class,'nav-item')]//button[contains(@title,'${key_webContentTitle}')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>GRAPHQL</td>
+	<td>//button[contains(.,'GraphQL')]</td>
+	<td></td>
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Button.path:996]: maybe we could join the GRAPHQL, HEADERS and ADD_HEADER by creating a generic locator with {value} to pass? And we would be passing the value in the macro or even in the step, example: `//button[contains(.,'${action}')]`

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Button.path [hunk 2]
```diff
@@ -990,6 +990,26 @@
...
+</tr>
+<tr>
+	<td>EXECUTE_QUERY</td>
+	<td>//button[contains(@title,'Execute Query (Ctrl-Enter)')]</td>
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Button.path:1000]: maybe we could add this one as well to the GRAPHQL, HEADERS and ADD_HEADER group of generic ones?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDiscovery/HeadlessDiscovery.testcase [hunk 3]
```diff
@@ -23,12 +57,14 @@ definition {
 		property portal.acceptance = "true";
 
 		task ("Given navigate to headless delivery endpoint") {
-			var portalURL = PropsUtil.get("portal.url");
-
-			Navigator.openSpecificURL(url = "${portalURL}/o/api?endpoint=${portalURL}/o/headless-delivery/v1.0/openapi.json");
+			APIExplorer.navigateToOpenAPI(
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDiscovery/HeadlessDiscovery.testcase:60]: nice 🎉

**Review by magjed4289 — CHANGES_REQUESTED** [review]

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Button.path (hunk 1 above)
Comment (yanan) [reply portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Button.path:996]: Great, can't agree more, I saw there are many //button[...contains...] in Button.path file, so I didn't combine these at the first time, will update :dancers:

**Comment by yanan** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
ci:test:relevant

**Comment by yanan** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
ci:test:relevant

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Button.path [hunk 4]
```diff
@@ -990,6 +990,11 @@
 	<td>//li[contains(@class,'nav-item')]//button[contains(@title,'${key_webContentTitle}')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>BUTTON_WITH_VALUE</td>
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uielements/Button.path:994]: 🎉

**Review by magjed4289 — APPROVED** [review]

**Comment by magjed4289** [comment]
ci:forward
